lib/net/ldap.rb

```diff
@@ -1286,11 +1286,9 @@ def use_connection(args)
     else
       begin
         conn = new_connection
-        if (result = conn.bind(args[:auth] || @auth)).result_code == Net::LDAP::ResultCodeSuccess
-          yield conn
-        else
-          return result
-        end
+        result = conn.bind(args[:auth] || @auth)
+        return result unless result.code == Net::LDAP::ResultCodeSuccess
```

Comment:
@tmaher I have just opened new PR to this your branch to fix this bug tmaher/ruby-net-ldap#1 . please check it.